### PR TITLE
[Bugfix] fix qdrant and weaviate db integration

### PR DESCRIPTION
**Pull request by deven298**
## Description

This PR fixes bugs related to qdrant and weaviate db integration. Specifically, it enables two functionalities with Qdrant and Weaviate.

- It skips embedding data when the content has not changed for the same source.
- Makes the query work with metadata filtering. For ex - 

```python
from embedchain import App

config = {
    "vectordb": {
        "provider": "qdrant",
        "config": {
            "collection_name": "my_qdrant_index"
        },
    }
}

app = App.from_config(config=config)

app.add("https://www.forbes.com/profile/elon-musk", metadata={"source": "forbes"})
app.add("https://en.wikipedia.org/wiki/Elon_Musk", metadata={"source": "wikipedia"})

response, context = app.query("which companies does elon own?", where={"source": "forbes"}, citations=True)
print(response)
print(list(map(lambda x: x[1]["url"], context)))
# The response will be coming from filtered documents where `source==forbes`.
```

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (does not change functionality, e.g. code style improvements, linting)
- [ ] Documentation update

## How Has This Been Tested?

- [x] Unit Test
- [x] Test Script (see above)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] Made sure Checks passed

### CODE REVIEW

File: embedchain/vectordb/qdrant.py [hunk 1]
```diff
@@ -82,21 +81,23 @@ def get(self, ids: Optional[list[str]] = None, where: Optional[dict[str, any]] =
...
 
-        qdrant_must_filters = [
-            models.FieldCondition(
-                key="identifier",
-                match=models.MatchAny(
-                    any=ids,
-                ),
+        qdrant_must_filters = []
+
+        if ids is not None and len(ids) > 0:
```
Comment (deven298) [inline embedchain/vectordb/qdrant.py:89]:
```suggestion
        if ids:
```

File: embedchain/vectordb/weaviate.py [hunk 2]
```diff
@@ -127,29 +129,64 @@ def get(self, ids: Optional[list[str]] = None, where: Optional[dict[str, any]] =
         :return: ids
         :rtype: Set[str]
         """
+        weaviate_where_operands = []
+
+        if ids is not None and len(ids) > 0:
```
Comment (deven298) [inline embedchain/vectordb/weaviate.py:134]:
```suggestion
        if ids:
```

File: embedchain/vectordb/weaviate.py [hunk 3]
```diff
@@ -127,29 +129,64 @@ def get(self, ids: Optional[list[str]] = None, where: Optional[dict[str, any]] =
...
                 .with_additional(["id"])
-                .with_limit(self.BATCH_SIZE),
-                cursor,
+                .with_limit(limit or self.BATCH_SIZE),
+                offset,
             )
+
             fetched_results = results["data"]["Get"].get(self.index_name, [])
-            if len(fetched_results) == 0:
+            if fetched_results is None or len(fetched_results) == 0:
```
Comment (deven298) [inline embedchain/vectordb/weaviate.py:177]:
```suggestion
            if not fetched_results:
```

**Review by taranjeet — APPROVED** [review]
this looks good. thanks

**Pull request merged**